pulp_container/app/management/commands/container-handle-image-data.py

```diff
@@ -31,41 +36,60 @@ class Command(BaseCommand):
     def handle(self, *args, **options):
         manifests_updated_count = 0
 
-        manifests = Manifest.objects.filter(labels={}, annotations={})
+        manifests = Manifest.objects.filter(Q(data="") | Q(annotations={}, labels={}))
         manifests = manifests.exclude(
             media_type__in=[MEDIA_TYPE.MANIFEST_LIST, MEDIA_TYPE.INDEX_OCI, MEDIA_TYPE.MANIFEST_V1]
         )
         manifests_updated_count += self.update_manifests(manifests)
 
         manifest_lists = Manifest.objects.filter(
-            media_type__in=[MEDIA_TYPE.MANIFEST_LIST, MEDIA_TYPE.INDEX_OCI], annotations={}
+            Q(media_type__in=[MEDIA_TYPE.MANIFEST_LIST, MEDIA_TYPE.INDEX_OCI]),
+            Q(data="") | Q(annotations={}),
         )
         manifests_updated_count += self.update_manifests(manifest_lists)
 
         self.stdout.write(
             self.style.SUCCESS("Successfully updated %d manifests." % manifests_updated_count)
         )
 
+    def init_manifest(self, manifest):
+        has_initialized_data = manifest.data != ""
+        if has_initialized_data:
+            manifest_data = json.loads(manifest.data)
+        else:
+            manifest_artifact = manifest._artifacts.get()
+            manifest_data, raw_bytes_data = get_content_data(manifest_artifact)
+            manifest.data = raw_bytes_data.decode("utf-8")
+            manifest._artifacts.clear()
```

Comment:
We are clearing the relation between the artifact and content. The removal can be handled afterwards by the orphan cleanup. Do you think we should erase the artifact immediately?

Reply:
The problem is that the very same artifact can be referenced from other plugins. We do not have control over the state of the things here. I think a safe bet is to remove just the relation. Can we agree on that?

Reply:
In that case a protected error would be raised. I am not going to insist on artifact removal here because they are pretty small, so we can keep them for orphan clean up.